packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -96,7 +96,7 @@ function clipFillAdjustedOutput(fill: Fill, remainingInput: BigNumber): BigNumbe
         return fill.adjustedOutput;
     }
     const penalty = fill.adjustedOutput.minus(fill.output);
-    return fill.output.times(remainingInput.div(fill.input)).plus(penalty);
+    return remainingInput.times(fill.rate).plus(penalty);
```

Comment:
Why was this function modified? based on my understanding, it should not impact `nativeOrdersToPath` which is the core function that you changed

Reply:
I find it just easier to reason about. `input * rate = output`, versus scaling the previous output by the proportion of remaining vs total.